### PR TITLE
Redirect admin pages to login for unauth'ed users

### LINKED ISSUE #567: [Idea]: Redirect 404's in /admin/ to the login
### Describe your idea / enhancement

I purposely made admin links show the default 404 page for "security" reasons. I do now think its way better UX if we can use deep links to admin features and admins being asked to log in and since its all open source its not really a big security feature anyways.

It could also be a "not authorized" page with a login button.

### How would this idea / enhancement be beneficial?

Allow deep linking to admin features without showing an error page if the users are not logged in.

### PR DESCRIPTION
**Pull request by JonasFranke**
## Describe your changes

Fixes #567 

## Notes for testing your change

...

### CODE REVIEW

**Comment by JonasFranke** [comment]
Currently, I'm showing an 'unauthorized' text on the sign-in page.
@pjeweb do you want an extra page for that like you said in the issue?

File: apps/website/src/pages/auth/signin.tsx
Comment (pjeweb) [inline apps/website/src/pages/auth/signin.tsx:1]: Not sure I get what you are doing here. I think redirecting to the home page after login is still needed. I also do not really think adding another headline is what we want (see review comment).

**Review by pjeweb — CHANGES_REQUESTED** [review]
> Currently, I'm showing an 'unauthorized' text on the sign-in page. @pjeweb do you want an extra page for that like you said in the issue?

Yeah, I think it would be best to 
- redirect unauthenticated users to `/auth/signin` (no message needed) and 
- redirect authenticated users without permissions (unauthorized) to a new page at `/admin/unauthorized`.

**Comment by JonasFranke** [comment]
Sorry for not working on this. I had a lot going on..
I kinda fucked up using git that's why the commits are so weird. Think it's done now

**Comment by JonasFranke** [comment]
Sorry for the messy commit messages. If you have any idea on how to fix this mess, please let me know!

**Pull request closed without merging**